File: shared/csharp/src/interfaces/HandlesDataEvents.cs

using PebbleFields.PebbleObject;
namespace PebbleFields.Shared.Interfaces
{
	/// <summary>
	/// </summary>
	/**
	 * implementations can use desired channel technology, notification timings etc.
	 * might need to cooperate with extended client appControlBase (ie gae channel, socket.io.js, EventSource)
	 * @author rtsunoda
	 *
	 */
	public interface HandlesDataEvents {
		void registerDevice(string id, Pebble paths);
		void notifyDataChange(Pebble data, string path, string srcDeviceKey);
		void notifyAdd(Pebble data, string path, string srcDeviceKey);
		void notifyUpdate(Pebble data, string path, string srcDeviceKey);
		void notifyDelete(string path, string srcDeviceKey);
	}
}

File: shared/java/src/com/pebblefields/shared/interfaces/HandlesDataEvents.java
package com.pebblefields.shared.interfaces;

import org.pebblefields.pebbleobject.Pebble;

/**
 * implementations can use desired channel technology, notification timings etc.
 * might need to cooperate with extended client appControlBase (ie gae channel, socket.io.js, EventSource)
 * @author rtsunoda
 *
 */
public interface HandlesDataEvents {
	void registerDevice(String id, Pebble paths);
	void notifyDataChange(Pebble data, String path, String srcDeviceKey);
	void notifyAdd(Pebble data, String path, String srcDeviceKey);
	void notifyUpdate(Pebble data, String path, String srcDeviceKey);
	void notifyDelete(String path, String srcDeviceKey);
}
